Update tooling

Alexandru-Dumitru opened the pull request:
Tools added:
- prettier
- madge
- eslint

Tools removed:
- tslint

zdmullen reviewed (approved):
Looks good!

VitGottwald commented on .vscodeignore:
```diff
@@ -17,11 +17,10 @@ src/**
 .gitignore
```
I think it is better to do it the other way around - forbidding everything and explicitly including the content we want included in the final vsix. Like we did it in Bridge for Git Explorer.

VitGottwald commented on Use-Cases-DONT_COMMIT.md:
```diff
@@ -0,0 +1,3 @@
+- create profile with same name (error OK)
```
Looks like an accidentally committed file - at least according to its name.

VitGottwald commented on tsconfig.json:
```diff
@@ -12,26 +12,21 @@
+    "sourceMap": true,
+    "strict": true /* enable all strict type-checking options */,
+    /* Additional Checks */
+    "noImplicitAny": false
```
We should enable this ASAP.

Alexandru-Dumitru replied:
I agree. + others. Thanks for bringing it up.

VitGottwald commented on tsconfig.json:
```diff
@@ -12,26 +12,21 @@
+    "target": "es6",
+    "outDir": "out",
+    "lib": ["es7"],
+    "skipLibCheck": true,
```
And we should disable this ASAP.

Alexandru-Dumitru commented:
I have addressed some of the feedback. For the compilation options, we need a roadmap, because each action triggers errors that need to be addressed. Thanks @VitGottwald for the review ❤️
